project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/settings/MinimizationPluginState.kt

```diff
@@ -5,35 +5,60 @@ import org.plan.research.minimization.plugin.model.FunctionLevelStage
 import org.plan.research.minimization.plugin.model.MinimizationStage
 import org.plan.research.minimization.plugin.model.state.*
 
-import com.intellij.openapi.components.BaseState
+import com.intellij.openapi.components.*
 import com.intellij.util.xmlb.annotations.Property
 import com.intellij.util.xmlb.annotations.XCollection
+import org.jetbrains.annotations.NonNls
 
-class MinimizationPluginState : BaseState() {
-    var currentCompilationStrategy by enum<CompilationStrategy>(CompilationStrategy.GRADLE_IDEA)
+@State(
+    name = "org.intellij.sdk.settings.AppSettings",
+    storages = [Storage("ProjectMinimizationSettings.xml")],
+)
+class MinimizationPluginState : PersistentStateComponent<MinimizationPluginState.State>, BaseState() {
```

Comment:
1. Look, it shouldn't be a `PersistentStateComponent` because we have one (`MinimizationPluginSettings`)
2. The `@State` annotation isn't required here because it shouldn't be a place where we store settings. Please move this annotation into `MinimizationPluginSettings` because only `MinimizationPluginSettings` should be responsible for storing the state in the file system and IDEA itself.

All in all: this state should be almost the same as it was before this PR, all additional functionality (as well as stateObservable) should be added in `MinimizationPluginSettings`. 

Hint: You can use the provided in DM simplified [example](https://outsight.labs.jb.gg/6dc42a67-cd3d-4e27-9cb2-db3179d0fcdb) that shows how it should be organized